ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -113,28 +119,33 @@
       String.format("UPDATE %s SET %s = ? WHERE %s", NAMED_BLOBS, DELETED_TS, PK_MATCH);
 
   private final AccountService accountService;
-  private final Map<String, DataSource> dcToDataSource;
   private final String localDatacenter;
-  private final ExecutorService executorService;
+  private final PriorityQueue<String> remoteDatacenters;
+  private final RetryExecutor retryExecutor;
+  private final Map<String, TransactionExecutor> transactionExecutors;
   private final MySqlNamedBlobDbConfig config;
 
   MySqlNamedBlobDb(AccountService accountService, MySqlNamedBlobDbConfig config, DataSourceFactory dataSourceFactory,
-      String localDatacenter) {
+      String localDatacenter, ExecutorService scheduler) {
     this.accountService = accountService;
     this.config = config;
     this.localDatacenter = localDatacenter;
-    this.dcToDataSource = MySqlUtils.getDbEndpointsPerDC(config.dbInfo)
+    this.retryExecutor = new RetryExecutor((ScheduledExecutorService) scheduler);
```

Comment:
One thought I had is that the `RetryExecutor` does not require the `scheduler` in this case since there is no sleep time or backoff. It is just some extra overhead in this case.

We could make a change to accept `null` scheduler in `RetryExecutor`:

Or, do you think this is too  much special casing for this situation?
```
diff --git a/ambry-commons/src/main/java/com/github/ambry/commons/RetryExecutor.java b/ambry-commons/src/main/java/com/github/ambry/commons/RetryExecutor.java
index c3bac271c..43574772a 100644
--- a/ambry-commons/src/main/java/com/github/ambry/commons/RetryExecutor.java
+++ b/ambry-commons/src/main/java/com/github/ambry/commons/RetryExecutor.java
@@ -67,8 +67,15 @@ public class RetryExecutor {
           int waitTimeMs = policy.waitTimeMs(currAttempts);
           logger.info("{} of {} attempts failed, will keep retrying after a {} ms backoff. exception='{}'",
               currAttempts, policy.maxAttempts(), waitTimeMs, exception);
-          scheduler.schedule(() -> recursiveAsyncRetry(call, isRetriable, policy, userCallback, currAttempts), waitTimeMs,
-              TimeUnit.MILLISECONDS);
+          if (scheduler == null) {
+            if (waitTimeMs > 0) {
+              logger.warn("No scheduler to delay execution of retry by {} ms", waitTimeMs);
+            }
+            recursiveAsyncRetry(call, isRetriable, policy, userCallback, currAttempts);
+          } else {
+            scheduler.schedule(() -> recursiveAsyncRetry(call, isRetriable, policy, userCallback, currAttempts),
+                waitTimeMs, TimeUnit.MILLISECONDS);
+          }
         } else {
           logger.info("{} of {} attempts failed, completing operation. exception='{}'", currAttempts,
               policy.maxAttempts(), exception);
```

Reply:
Sure, updated in this way to handle cases when no sleep time need for retry logic.